packages/collectxyz/src/nft.rs

```diff
@@ -2,12 +2,55 @@ use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 use std::convert::TryInto;
 
-use cosmwasm_std::{Binary, Coin, StdError, StdResult, Timestamp, Uint128};
-use cw721::Expiration;
-use cw721_base::{
-    msg::{ExecuteMsg as CW721ExecuteMsg, QueryMsg as CW721QueryMsg},
-    state::TokenInfo,
-};
+use cosmwasm_std::{Addr, Binary, Coin, StdError, StdResult, Timestamp, Uint128};
+use cw721::{Expiration, OwnerOfResponse};
+use cw721_base::msg::{ExecuteMsg as CW721ExecuteMsg, QueryMsg as CW721QueryMsg};
+use cw721_base::state::Approval;
+
+// ----------------- begin CW721 ^0.9.2 shim ----------------- //
+
+// adapted from: https://github.com/CosmWasm/cw-nfts/blob/5e1e72a3682f988d4504b94f2e203dd4a5a99ad9/contracts/cw721-metadata-onchain/src/lib.rs#L7-L26
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug, Default)]
+pub struct Cw721Trait {
+    pub display_type: Option<String>,
+    pub trait_type: String,
+    pub value: String,
+}
+
+// see: https://docs.opensea.io/docs/metadata-standards
```

Comment:
Is this available from a crate anywhere?

Reply:
Only from [this crate](https://github.com/CosmWasm/cw-nfts/tree/main/contracts/cw721-metadata-onchain) to my knowledge, which has peer dependencies on cw721 and cw721-base v0.9.2 that conflict with our dependencies on v0.9.1.

Of course, copy-and-paste isn't ideal, but given that this metadata standard will be pretty stable, I think this is a fine interim solution while we develop a plan to migrate off of cw721 v0.9.1.

Reply:
Put it in notion! To do that upgrade